### PR TITLE
[Spark]fix update bug

### PR DESCRIPTION
**Pull request by hellojinsilei**
## Why are the changes needed?
Fix the following bugs
1.Updating column value with decimal、double and float type will fail in Spark
2.Updating data across multiple partitions will fail in Spark

## Brief change log

  - *Fix bugs related to Spark SQL Update*

## How was this patch tested?
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

### CODE REVIEW

File: core/src/main/java/com/netease/arctic/io/writer/SortedPosDeleteWriter.java [hunk 1]
```diff
@@ -80,6 +80,17 @@ public SortedPosDeleteWriter(FileAppenderFactory<T> appenderFactory,
     this(appenderFactory, fileFactory, format, mask, index, partitionKey, DEFAULT_RECORDS_NUM_THRESHOLD);
   }
 
+  public SortedPosDeleteWriter(FileAppenderFactory<T> appenderFactory,
+                               OutputFileFactory fileFactory,
+                               FileFormat format,
+                               StructLike partitionKey) {
+    this.appenderFactory = appenderFactory;
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/io/writer/SortedPosDeleteWriter.java:87]: How about use `this(appenderFactory, fileFactory, format, 0, 0, partitionKey, DEFAULT_RECORDS_NUM_THRESHOLD);` instead.

**Review by zhoujinsong — COMMENTED** [review]
@zhoujinsong I left some comment. PTAL.

File: core/src/main/java/com/netease/arctic/io/writer/SortedPosDeleteWriter.java (hunk 1 above)
Comment (hellojinsilei) [reply core/src/main/java/com/netease/arctic/io/writer/SortedPosDeleteWriter.java:87]: Using this constructor, the generated POS-delete file has the wrong name

Comment (zhoujinsong) [reply core/src/main/java/com/netease/arctic/io/writer/SortedPosDeleteWriter.java:87]: Okay, but I donnot think using a `NULL` data node is a good policy. We could improve this in another PR.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**